frontend_tests/casper_tests/00-realm-creation.js

```diff
@@ -0,0 +1,91 @@
+var common = require('../casper_lib/common.js').common;
+
+var email = 'alice@example.com';
+var domain = 'example.com';
+var organization_name = 'Awesome Organization';
+
+casper.start('http://localhost:9981/create_realm/');
+
+casper.then(function () {
+    // Submit the email for realm creation
+    this.waitForSelector('form[action^="/create_realm/"]', function () {
+        this.fill('form[action^="/create_realm/"]', {
+            email: email
+        }, true);
+    });
+    // Make sure confirmation email is send
+    this.waitWhileSelector('form[action^="/create_realm/"]', function () {
+         var regex = new RegExp('^http:\/\/[^\/]+\/accounts\/send_confirm\/' + email);
+         this.test.assertUrlMatch(regex, 'Confirmation mail send');
+    });
+});
+
+// Special endpoint enabled only during tests for extracting confirmation key
+casper.thenOpen('http://localhost:9981/confirmation_key/');
```

Comment:
@timabbott I don't think casper can do that without navigating. They have given an example on retrieving JSON file which uses the same method. 
http://docs.casperjs.org/en/latest/modules/casper.html#getpagecontent

But I think I can replace that step with a pure node.js function if that is necessary.